### PR TITLE
quassel: Add separate quassel-core package

### PR DESCRIPTION
**Pull request by Biswa96**
Fixes https://github.com/msys2/MINGW-packages/issues/12967

### CODE REVIEW

**Comment by MehdiChinoune** [comment]
I don't see any new separate package?

**Comment by MehdiChinoune** [comment]
Sorry, Isn't quassel-core contained in quassel?.
What's the problem this PR is trying to solve?. Is there an issue with `quassel`?
If we follow user requests we will end up like debian/redhat packaging system (runtime + plugins + development + debug + docs).
I was expecting two split packages where `quassel` depends on `quassel-core`. This approach just wastes buildtime for no valid reason.

**Comment by Biswa96** [comment]
> Isn't quassel-core contained in quassel?

Yes, that's what I first thought of. But then I follow what ArchLinux does. I think this approach is more flexible for adding any further configuration (like Arch does) in future.

File: mingw-w64-quassel/PKGBUILD [hunk 1]
```diff
@@ -36,9 +38,12 @@ prepare() {
 }
 
 build() {
-  cd "${srcdir}"
-  [[ -d build-${CARCH} ]] && rm -rf build-${CARCH}
-  mkdir build-${CARCH} && cd "${srcdir}"/build-${CARCH}
```
Comment (mmuetzel) [inline mingw-w64-quassel/PKGBUILD:41]: Would it make sense to continue making sure that the build directory is clean?

Comment (Biswa96) [reply mingw-w64-quassel/PKGBUILD:41]: See https://github.com/msys2/MINGW-packages/commit/10700daef1e47f6f842b4e987c0f2baebd59e6af

**Pull request closed without merging**